File: lib/mocked/mocke3/src/main/java/com/mobiata/mocke3/FileSystemOpener.java
package com.mobiata.mocke3;

import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;

public class FileSystemOpener implements FileOpener {
	private String mRoot;

	public FileSystemOpener(String root) {
		mRoot = root;
	}

	@Override
	public InputStream openFile(String filename) throws IOException {
		InputStream inputStream = new FileInputStream(mRoot + "/" + filename);
		return inputStream;
	}
}

File: lib/mocked/mocke3/src/main/kotlin/com/mobiata/mocke3/FileSystemOpener.kt
package com.mobiata.mocke3

import java.io.FileInputStream
import java.io.IOException
import java.io.InputStream

public class FileSystemOpener(private val root: String) : FileOpener {
	@Throws(IOException::class)
	override fun openFile(filename: String): InputStream {
		return FileInputStream(root + "/" + filename)
	}
}
